src/components/views/rooms/wysiwyg_composer/hooks/useSuggestion.ts

```diff
@@ -17,25 +17,37 @@ limitations under the License.
 import { Attributes, MappedSuggestion, SuggestionChar, SuggestionType } from "@matrix-org/matrix-wysiwyg";
 import { SyntheticEvent, useState } from "react";
 
-// This type is a close approximation of what we use in the Rust model for the rich version of the
-// editor, we use this to try and make this simple model as similar as possible to allow reuse of
-// the autocomplete component
+/* To allow reuse of the autocomplete component, we need to record similar information as used in 
+the Rust model (keyChar, type and text) as well as information required to allow us to do the 
+DOM manipulation when replacing the suggestion with a mention or command (node, start and
+end offsets). */
 export type PlainTextSuggestionPattern = {
     keyChar: SuggestionChar;
     type: SuggestionType;
     text: string;
     node: Node;
     startOffset: number;
     endOffset: number;
-} | null;
+};
+type Suggestion = PlainTextSuggestionPattern | null;
 
 /**
- * Given an input div element, return the current suggestion found in that div element if present
- * as well as handlers for command or mention completions from the autocomplete, and a listener
- * to be attached to the PlainTextComposer
+ * React hook to allow tracking and replacing of mentions and commands in a div element
  *
  * @param editorRef - a ref to the div that is the composer textbox
+ * @param setText - setter function to allow the hook to update the content state when replacing
+ * text with text selected from the autocomplete
+ * @returns
+ * - `handleMention`: TODO a function that will allow @ or # mentions to be selected from
+ * the autocomplete and inserted into the composer
+ * - `handleCommand`: a function that allows slash commands selected from the autocomplete to be inserted
+ * into the composer
```

Comment:
The use of "allows" here is misleading, IMHO - it suggests that this function can be used as part of the process of doing inserting slash commands, but doesn't really explain how it fits into that process. Let's be more explicit:

```suggestion
 * - `handleCommand`: a function that will replace the content of the composer with the given replacement text. Can be used to process autocomplete of slash commands.
```